types/detect/detect.go

```diff
@@ -38,7 +38,34 @@ type SignatureEventSelector struct {
 type SignatureHandler func(found Finding)
 
 //Event is a generic event that the Engine can process
-type Event interface{}
+type EventHeaders struct {
+	// ContentType indicates the content of the Event payload.
+	ContentType string
+	// Origin indicates whether Event originates from host or container.
+	Origin string
+	// Headers additional custom headers, nil most of the time
+	Headers map[string]string
```

Comment:
Just an idea to call these headers `Custom` to distinguish from mandatory ContentType and Origin. WDYT?

```go
event.Headers.CotentType
event.Headers.Custom["Content-Length"]
```

Reply:
I like it, i'll add that change.